pkg/cloudevents/transport/http/message.go

```diff
@@ -77,3 +81,32 @@ func (m Message) CloudEventsVersion() string {
 
 	return ""
 }
+
+func readAllClose(r io.ReadCloser) ([]byte, error) {
+	if r != nil {
+		defer r.Close()
+		return ioutil.ReadAll(r)
+	}
+	return nil, nil
+}
+
+// FromRequest initializes a Message from a http.Request
+func (m *Message) FromRequest(r *http.Request) (err error) {
+	copyHeadersEnsure(r.Header, &m.Header)
+	m.Body, err = readAllClose(r.Body)
+	return err
+}
+
+// ToRequest copies a message to a http.Request.
+// Replaces r.Body, r.ContentLength and r.Method.
+// Updates r.Headers.
+func (m *Message) ToRequest(r *http.Request) {
+	copyHeadersEnsure(m.Header, &r.Header)
```

Comment:
r could be nil

Reply:
That would panic. I should state that in the docs, but using a pointer as a way to provide "write access" to a parameter that is not allowed to be nil is a pretty common idiom, since Go has no other way to do that.

On a more general note, I see a lot of func (x *X) { if x == nil...} in the CE code. There are rare cases where it makes sense to operate on a nil pointer but they are... rare. IMO in most cases, calling a method on a nil pointer is a logic error and should result in an immediate panic, not get buried in a log file (if you're lucky) as a run-time error while the program to continues rolling towards certain doom. It's a debatable point, I like a good debate :)